### PR TITLE
Added code for Array Manipulaiton in C++

### PR DESCRIPTION
**Pull request by yr7000**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description
Added the code for Array manipulation in C++ in the cp folder of the c-plus-plus section.

### Checklist

- [ yes] I've read the contribution guidelines.
- [ yes] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Thanks for assigning this issue(#3027  ) to me.

### CODE REVIEW

**Comment by yr7000** [comment]
Hey @HarshCasper ,  please review this pr.

File: C-Plus-Plus/cp/Array_Manipulation.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Time Complexity  : O(n)
+Space Complexity : O(n)
+Input: 
+5 3
+1 2 100
+2 5 100
+3 4 100
+Output:
+200 
+*/
```
Comment (bhav09) [inline C-Plus-Plus/cp/Array_Manipulation.cpp:55]: Add a new at EOF.

File: C-Plus-Plus/cp/Array_Manipulation.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+#define ll long long int
+
+int main()
+{
+    ll n, m, a, b, k;
+    cin >> n >> m;
+    vector<ll> prefix_sum(n + 1, 0);
+    for (ll queries = 1; queries <= m; queries++)
```
Comment (bhav09) [inline C-Plus-Plus/cp/Array_Manipulation.cpp]: Your main function should be the driver function. Make sure you write the logic of the code in a separate function and use main function just for calling the function.

File: C-Plus-Plus/cp/Array_Manipulation.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        cin >> a >> b >> k;
+        prefix_sum[a] += k;
+        if ((b + 1) <= n)
+        {
+            prefix_sum[b + 1] -= k;
+        }
+    }
+    ll answer = INT_MIN;
+    for (ll index = 1; index <= n; index++)
```
Comment (bhav09) [inline C-Plus-Plus/cp/Array_Manipulation.cpp]: Your code lacks comments heavily.

File: C-Plus-Plus/cp/Array_Manipulation.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        prefix_sum[index] += prefix_sum[index - 1];
+        answer = max(answer, prefix_sum[index]);
+    }
+    cout << answer << endl;
+}
+
+/*
+Time Complexity  : O(n)
```
Comment (bhav09) [inline C-Plus-Plus/cp/Array_Manipulation.cpp:46]: Good. Perfectly written.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

File: C-Plus-Plus/cp/Array_Manipulation.cpp (hunk 1 above)
Comment (yr7000) [reply C-Plus-Plus/cp/Array_Manipulation.cpp:55]: Should I add a new line at the end of file?

**Comment by yr7000** [comment]
Hey @bhav09, thanks for pointing out the mistakes. I made all the changes, please review this pr and please give feedback on whether it is fine and is it ready to merge.
Thank you.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**